libs/mock-block-dock/src/mock-block-dock-context.tsx

```diff
@@ -17,7 +17,7 @@ import {
 } from "react";
 
 type MockBlockDockInfo = {
-  blockEntitySubgraph: EntityRootedSubgraph;
+  blockEntitySubgraph: Subgraph<SubgraphRootTypes["entity"]>;
```

Comment:
This really confuses me.. Any ideas @CiaranMn

Reply:
It looks like TypeScript does not or cannot infer the generic `RootType` from the `subgraph` argument when the declared type of the argument passed hides the generic behind an alias (as the `EntityRootedSubgraph` does). I had a try to google existing issues to do with this but didn't get anywhere.

<img width="696" alt="Screenshot 2023-01-31 at 15 31 08" src="https://user-images.githubusercontent.com/37743469/215804669-9797ab5a-9df2-4010-9b14-8e515f316bed.png">

Because of that, I wonder if it is better not having these aliases (sorry) so that we don't cause ourselves or others this pain. A slight improvement over the previous position would be to split out the root types so that we could just do `Subgraph<EntityRootType>` rather than `Subgraph<SubgraphRootTypes["entity"]>`

```ts

export type EntityRootType = {
  vertexId: EntityVertexId;
  element: Entity;
};

export type EntityTypeRootType = {
  vertexId: OntologyTypeVertexId;
  element: EntityTypeWithMetadata;
};

export type DataTypeRootType = {
  vertexId: OntologyTypeVertexId;
  element: DataTypeWithMetadata;
};

export type PropertyTypeRootType = {
  vertexId: OntologyTypeVertexId;
  element: PropertyTypeWithMetadata;
};

export type SubgraphRootType =
  | EntityRootType
  | EntityTypeRootType
  | DataTypeRootType
  | PropertyTypeRootType;
```

Reply:
I was able to recreate this in a _somewhat_ simple form on a [TS playground](https://www.typescriptlang.org/play?#code/PTAEEsHMDsHsCcCmoAuALZ5rUfVBPAB0QGdQMkAaVDfAciVBIFdi8BjWaFecAN0QATUADMEoAIZM0CFKEQAPCQFtCAG0QAoFEWQBRDcsTcAKrrIBeUAG9NoVIpQAuG3fsRBLgEaxYGidDUbvZILGrOTDxYkG4AvpRuWCiIkLgutu4eLrBeAFaI7CgJmaHM4S7QzMpeuHGasZqaioSyBMSgBohGprqgVp3dKGbEJADaANaI+LAiHYbGQ+YAuo067QBivgA8JvIKydCCZAMLw4gAfH2u9uCeoCajAES3jyv2IKAAdN-1jR8mGBIyHYEmYQNAhHgOUMZCwoAABl4JPB4dpeiZHJtYFcsVsTj0Rk9kvtXuc0e0AJLcFK4LE47b4xaE57U1LwUmNTjQEhyJF4Kw7PYHI5zLqnXTnAAUYlgLlxJnOAEoXA9HqVwq8+pdrKAkChmPBoKAAAySMgBfAAblAsVAH3Aqnm3AkKHAXFAsGU4BQyWEnK4PIc+wAysR2OAROAhHKGU6maQiY5SVcdbcXCIJGpwQ0uYHiSgAIJqcASEgqzG+FNZUSZ7OaXNyJI0+ChgoRqN3XGMs5jFnJNnJqypu7WW05gON1m4IslsugKn92mVofV0c2-3cuQkMPtoQYnkARiufMl+db4cjQkV1o+nHgSEKanwbWQPN40BiDYcPJnpaPVhPfNfxIa87TALA7wfFAnxfSJ30gUAAB9QEqapai-bc20vQR9xQAAmY9kUlJs2XPXdBFA28ECgmC1mQVCangesJ2-Qti1LAiAKIkjp3YkCb3A6BIIKaDnzouDoiQlCqkYoA)

I'm going to ask on the TS discord and see if anyone has ideas. Your alternative posed here does seem like an improvement on what we had before though, I'll probably go with that.

Reply:
Seems that this is to do with how TS evaluates types and its reliance on structural typing ([Discord thread](https://discord.com/channels/508357248330760243/1070295766930817024/1070295766930817024)).

I've opted to go with the suggestion you gave.